src/lib/routes/util.ts

```diff
@@ -13,7 +13,11 @@ export const customJoi = joi.extend((j) => ({
     },
     validate(value, helpers) {
         // Base validation regardless of the rules applied
-        if (encodeURIComponent(value) !== value) {
+        if (
+            encodeURIComponent(value) !== value ||
+            value === '..' ||
+            value === '.'
```

Comment:
Yeh, this is good. Are there other cases we might have missed? 

I checked the docs for this function, and there's also `- _ ! ~ * ' ( )`:

![image](https://github.com/user-attachments/assets/3bcb2982-7744-4fd2-b545-5a3f48939175)

Do we want to allow them? I guess there's no harm in it because they don't break urls the same way. Sure, let's go.